### ISSUE TITLE
Automate the client's makefile so that we don't have to modify it every time

### ISSUE THREAD

**Issue by pblcc** [pblcc/trj#4]
The makefile of the client looks like:
```makefile
TARGET = client
LIBS = -pthread
CC = gcc
CFLAGS = -g
FILES = # The files to be compiled

client: $(FILES)
	gcc -o $(TARGET) $(LIBS) $(FILES) $(CFLAGS)
```
The problem with this is that we have to change the `FILES` variable every time we have to add a new file, so it would be great if this could be automated in the makefile.
Remember that the trojan is in a different directory from the client itself and it also has to be compiled.
```
/trj
| ...
| /client
|| ...
||/trojan
||| ...
```
##### Read the contribution guidelines to resolve this issue.

**Comment by mpzaborski** [comment]
Hello!

See [#5](https://github.com/pblcc/trj/pull/5) as a proposal for issue.

> it would be great if this could be automated in the makefile

I adjusted the client hierarchy to make it possible - there was a main() collision between client.c and ntrj.test.c so I moved the test related files to separate folder (Unless you use some unit test framework it is better to keep them in separate directory if you call test by main())

I also applied changes in makefiles to allow for building test ntrj.c , Initially the test lacked libraries on my platform and had typo that prevented it from building.

I hope it will help you in further development. 

Regards,
Mateusz
